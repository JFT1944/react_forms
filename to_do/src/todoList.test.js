import React from "react";
import { render, fireEvent, getByTestId } from "@testing-library/react";
import  ToDoList from "./TodoList";
// import TEST_IMAGES from "./_testCommon.js";
import App from './App'



it("renders without crashing", async function() {
    render(<ToDoList  />);
  });


it("matches snapshot", function() {
    const {asFragment} = render(<ToDoList />);
    expect(asFragment()).toMatchSnapshot();
  });
  

// test a button click
it('should delete a todo', function(){
    let {getByTestId, getByText} = render(<ToDoList />)
    let deleteB = getByTestId('subBut1')
    expect(deleteB).toBeInTheDocument()
    fireEvent.click(deleteB)
    expect(deleteB).not.toBeInTheDocument()
}

)
it('should add a todo', function(){
    let {getByTestId, getByText} = render(<ToDoList />)
    let inputForm = getByTestId('input_form')
    expect(inputForm).toBeInTheDocument()
    fireEvent.change(inputForm, {target: {value: 'test'}})
    let addB = getByTestId('subBut')
    expect(addB).toBeInTheDocument()
    fireEvent.click(addB)
    let newItem = getByText('test')
    expect(newItem).toBeInTheDocument()
}

)
