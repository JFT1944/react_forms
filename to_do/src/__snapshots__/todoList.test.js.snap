// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <input
      data-testid="input_form"
      type="text"
      value=""
    />
    <input
      data-testid="subBut"
      type="submit"
    />
  </form>
  <ol>
    <li>
      Feed the Velociraptor 
      <button
        data-testid="subBut0"
      >
        X
      </button>
    </li>
    <li>
      Pet the Lobster 
      <button
        data-testid="subBut1"
      >
        X
      </button>
    </li>
  </ol>
</DocumentFragment>
`;
